From 4acbd9a96b7b3c4531d70544541574a2d5bd5538 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Thu, 18 Feb 2021 16:36:23 +0200
Subject: [PATCH] [SW-20154] gaudi2_arc_gnu: increase the stack to 8kb

Gaudi2 has a new feature called eROM, wherein the
ROM code can possibly jump into different binary and
return. Hence this needs a common stack usage between
ROM and other eROM bins. Also since the key hash size is
increased to 384 bits, the arrays declared on the stack
by the common framework will increase by 4 DWORDS
per key. Hence to meet these requirements, we increase
the stack size.

Change-Id: Ied2a64595ccc5ae08c88ca5f88dc3045de5f7a58
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/gaudi2_arc_gnu/arcboot_rom.lds b/gaudi2_arc_gnu/arcboot_rom.lds
index 1721eb2..55904dd 100644
--- a/gaudi2_arc_gnu/arcboot_rom.lds
+++ b/gaudi2_arc_gnu/arcboot_rom.lds
@@ -90,7 +90,7 @@
 	.stack : {
 		. = ALIGN(32 / 8);
 		__stack = . ;
-		. = . + (DEFINED(__STACK_SIZE) ? __STACK_SIZE : (DEFINED(__DEFAULT_STACK_SIZE) ? __DEFAULT_STACK_SIZE : 2k)) ;
+		. = . + (DEFINED(__STACK_SIZE) ? __STACK_SIZE : (DEFINED(__DEFAULT_STACK_SIZE) ? __DEFAULT_STACK_SIZE : 8k)) ;
 		__stack_top = . ;
 	} > REGION_STACK
 }
